src/handlers/strategies/directoryListing.ts

```diff
@@ -97,6 +104,55 @@ export function renderDirectoryListing(
   });
 }
 
+/**
+ * Send a request to R2 to get the objects & paths in a directory
+ * @param client {@link S3Client} to use for the request
+ * @param bucketPath Path in R2 bucket
+ * @param cursor Where to begin the request from, for pagination
+ * @param env Worker env
+ * @returns A {@link ListObjectsV2CommandOutput}
+ * @throws When all retries are exhausted and no response was returned
+ */
+async function fetchR2Result(
+  client: S3Client,
+  bucketPath: string,
+  cursor: string | undefined,
+  env: Env
+): Promise<ListObjectsV2CommandOutput> {
+  let result: ListObjectsV2CommandOutput | undefined = undefined;
+
+  let retriesRemaining = S3_RETRY_LIMIT;
+  while (retriesRemaining > 0) {
+    try {
+      // Send request to R2
+      result = await client.send(
+        new ListObjectsV2Command({
+          Bucket: env.BUCKET_NAME,
+          Prefix: bucketPath,
+          Delimiter: '/',
+          MaxKeys: 1000,
+          ContinuationToken: cursor,
+        })
+      );
+
+      // Request succeeded, no need for any retries
+      break;
```

Comment:
This should simply return the result right here, so that you don't need to make the if statement below.

Pretty much:

```
return await client.send()...
```